sys/event/event.c

```diff
@@ -110,6 +114,30 @@ event_t *event_wait(event_queue_t *queue)
     return result;
 }
 
+#ifdef MODULE_XTIMER
+event_t *event_wait_until(event_queue_t *queue, uint32_t timeout)
```

Comment:
As discussed offline: please rename to "event_wait_timeout()", to be more inline with msg_receive_timeout, xtimer_mutex_lock_timeout(), and also to not make anyone think "until" is an absolute time.